pkg/mountutil/mountutil.go

```diff
@@ -54,73 +46,14 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
 		res      Processed
 		src, dst string
 		options  []string
+		err      error
 	)
 
-	split := strings.Split(s, ":")
-	switch len(split) {
-	case 1:
-		dst = s
-		res.AnonymousVolume = idgen.GenerateID()
-		logrus.Debugf("creating anonymous volume %q, for %q", res.AnonymousVolume, s)
-		anonVol, err := volStore.Create(res.AnonymousVolume, []string{})
-		if err != nil {
-			return nil, err
-		}
-		src = anonVol.Mountpoint
-		res.Type = Volume
-	case 2, 3:
-		res.Type = Bind
-		src, dst = split[0], split[1]
-		if !strings.Contains(src, "/") {
-			// assume src is a volume name
-			res.Name = src
-			vol, err := volStore.Get(src)
-			if err != nil {
-				if errors.Is(err, errdefs.ErrNotFound) {
-					vol, err = volStore.Create(src, nil)
-					if err != nil {
-						return nil, err
-					}
-				} else {
-					return nil, err
-				}
-			}
-			// src is now full path
-			src = vol.Mountpoint
-			res.Type = Volume
-		}
-		if !filepath.IsAbs(src) {
-			logrus.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", src)
-			var err error
-			src, err = filepath.Abs(src)
-			if err != nil {
-				return nil, err
-			}
-		}
-		if !filepath.IsAbs(dst) {
-			return nil, fmt.Errorf("expected an absolute path, got %q", dst)
-		}
-		rawOpts := ""
-		if len(split) == 3 {
-			rawOpts = split[2]
-		}
-		res.Mode = rawOpts
-
-		// always call parseVolumeOptions for bind mount to allow the parser to add some default options
-		var err error
-		var specOpts []oci.SpecOpts
-		options, specOpts, err = parseVolumeOptions(res.Type, src, rawOpts)
-		if err != nil {
-			return nil, err
-		}
-		res.Opts = append(res.Opts, specOpts...)
-	default:
-		return nil, fmt.Errorf("failed to parse %q", s)
+	src, dst, options, err = ProcessSplit(s, volStore, &res, src, dst, options)
```

Comment:
Looking at the implementations, `src`, `dst`, and `options` are never consumed, so should probably not be parameters, but local vars in `ProcessSplit`. That would allow simplifying this call slightly to `src, dst, options, err := ProcessSplit(s, volStore, &res)`, trimming the top-of-function `var` lines, rather than growing them.

Also, does it need to be public? `processSplit` seems like it should work, all uses are in pkg/mountutil AFAICT, and it seems slightly safer to hide platform-split functions to control the blast radius in case of future divergence.